tests/integrations/cohere/test_cohere.py

```diff
@@ -0,0 +1,202 @@
+
+    with start_transaction(name="cohere tx"):
+        response = client.chat(
+            model="some-model",
+            chat_history=[ChatMessage(role="SYSTEM", message="some context")],
+            message="hello",
+        ).text
+
+    assert response == "the model response"
+    tx = events[0]
+    assert tx["type"] == "transaction"
+    span = tx["spans"][0]
+    assert span["op"] == "ai.chat_completions.create.cohere"
+    assert span["data"]["ai.model_id"] == "some-model"
+
+    if send_default_pii and include_prompts:
+        assert "some context" in span["data"]["ai.input_messages"][0]["content"]
+        assert "hello" in span["data"]["ai.input_messages"][1]["content"]
+        assert "the model response" in span["data"]["ai.responses"]
+    else:
+        assert "ai.input_messages" not in span["data"]
+        assert "ai.responses" not in span["data"]
+
+    assert span["measurements"]["ai_completion_tokens_used"]["value"] == 10
+    assert span["measurements"]["ai_prompt_tokens_used"]["value"] == 20
+    assert span["measurements"]["ai_total_tokens_used"]["value"] == 30
+
+
+# noinspection PyTypeChecker
+@pytest.mark.parametrize(
+    "send_default_pii, include_prompts",
+    [(True, True), (True, False), (False, True), (False, False)],
+)
+def test_streaming_chat(sentry_init, capture_events, send_default_pii, include_prompts):
+    sentry_init(
+        integrations=[CohereIntegration(include_prompts=include_prompts)],
+        traces_sample_rate=1.0,
+        send_default_pii=send_default_pii,
+    )
+    events = capture_events()
+
+    client = Client(api_key="z")
+    HTTPXClient.send = mock.Mock(
+        return_value=httpx.Response(
+            200,
+            content="\n".join(
+                [
+                    json.dumps({"event_type": "text-generation", "text": "the model "}),
+                    json.dumps({"event_type": "text-generation", "text": "response"}),
+                    json.dumps(
+                        {
+                            "event_type": "stream-end",
+                            "finish_reason": "COMPLETE",
+                            "response": {
+                                "text": "the model response",
+                                "meta": {
+                                    "billed_units": {
+                                        "output_tokens": 10,
+                                        "input_tokens": 20,
+                                    }
+                                },
+                            },
+                        }
+                    ),
+                ]
+            ),
+        )
+    )
+
+    with start_transaction(name="openai tx"):
+        responses = list(
+            client.chat_stream(
+                model="some-model",
+                chat_history=[ChatMessage(role="SYSTEM", message="some context")],
+                message="hello",
+            )
+        )
+        response_string = responses[-1].response.text
```

Comment:
only the last response is important?

Reply:
cohere sends a summary response at the end of the stream that contains the entire message content.